### DISCUSSION TITLE
Multi-site sync for author profiles

### DISCUSSION THREAD

**Discussion by stevejburge** [publishpress/PublishPress-Authors#1226] [Feature Requests]
> We are running a multisite, hence user profile/data are shared across sites but author profile/data are site specific. And our students will be using these multiple (course) sites. We want to add fields, which you document here for authors, but we'd like to do so with user profiles as the "mother ship" of sorts in our multisite. Otherwise students would need to add redundant user information on each site.

**Comment by andergmartins** [comment]
I still need to work on this, but here is a code I sent for a user to test until we implement this as a feature in our plugin. Btw, need more testing for a multisite. 

```
use MultipleAuthors\Classes\Objects\Author;

add_action('admin_init', 'setAuthorsSyncCronJob');
add_action('custom_pp_authors_sync_users', '_ppAuthorsSyncUsers');

function setAuthorsSyncCronJob()
{
    if (!wp_next_scheduled('custom_pp_authors_sync_users')) {
        wp_schedule_event(time(), 'hourly', 'custom_pp_authors_sync_users');
    }
}


function _ppAuthorsSyncUsers()
{
    $authors = get_terms(
        'author',
        [
            'hide_empty' => false,
        ]
    );

    foreach ($authors as $authorTerm) {
        $author = Author::get_by_term_id($authorTerm->term_id);

        if ($author->is_guest()) {
            continue;
        }

        Author::update_author_from_user($author->term_id, $author->user_id);
    }
}
```

Maybe having a way to filter which user should be synced would be good (by user role?)
